pkg/kube/translation/annotations/redirect.go

```diff
@@ -15,11 +15,15 @@
 package annotations
 
 import (
+	"strconv"
+
 	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
 )
 
 const (
-	_httpToHttps = AnnotationsPrefix + "http-to-https"
+	_httpToHttps           = AnnotationsPrefix + "http-to-https"
+	_permanentRedirect     = AnnotationsPrefix + "permanent-redirect"
+	_permanentRedirectCode = AnnotationsPrefix + "permanent-redirect-code"
```

Comment:
Do you mean to add test cases for other status codes?

Reply:
I mean, we should only allow users configuring `301` or `308` here. If I configure a `302` code here, it doesn't comply the HTTP semantic, because `302` is a "temporarily" redirect, but what we want is "permanent".

Reply:
Got it.

Reply:
@tokers  @AlinsRan  According to [RFC 7231](https://www.rfc-editor.org/rfc/rfc7231.html)  Status codes between 300 and 308 can indicate redirection. 

Considering that we will also support temporary redirection, but adding a specific implementation for it alone is not of much value, here I suggest to merge it directly and modify this annotation to `http-redirect` and `http-redirect-code` , the corresponding verification logic is, if not configured, the default is `http.StatusMovedPermanently`, the allowed value is between `http.StatusMultipleChoices` and `http.StatusPermanentRedirect`.